src/OrchardCore.Modules/OrchardCore.Queries/Controllers/AdminController.cs

```diff
@@ -155,13 +160,15 @@ public async Task<IActionResult> CreatePost(QueriesCreateViewModel model)
             {
                 return NotFound();
             }
-
+            if (!string.IsNullOrWhiteSpace(query.Schema) && !CheckSchema(query.Schema))
+            {
+                ModelState.AddModelError("Schema", S["Invalid schema JSON supplied."]);
```

Comment:
Ok great, this works better.

Sorry, what I should have said though, is to implement `IValidatableObject ` on the `QueriesCreateViewModel`.

Apply the test for valid JSON on that View Model by implementing `IValidatableObject`.

Here is an example from where it is used elsewhere in the code https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/ViewModels/LinkExternalLoginViewModel.cs

I think you will need to apply the same to the edit controller `QueriesEditViewModel` View Model

Then in the `Queries.Fields.Edit.cshtml` View you should be able to apply 
`<span asp-validation-for="Schema"></span>` (like is done for `Name` in the same file) and you will get a validation error on the schema field itself

Reply:
But the value of Schema comes from query variable, not model, model.Schema is always empty

Reply:
![image](https://user-images.githubusercontent.com/15613121/86607411-cc076080-bfdb-11ea-8e32-fbf039d30842.png)
And the query object cannot be filled if it is verified during creation

Reply:
Ok, I just added a string extension for testing json here https://github.com/OrchardCMS/OrchardCore/pull/6614

But I have had a look at the Queries `AdminController` and understand what is driving that viewmodel now.

I was wrong to suggest `IValidateableObject`.

The query `Schema`, is actually managed and updated by the `QueryDisplayDriver.cs`

It already performs some checks for validity so the test you are doing for valid Json should move into the `QueryDisplayDriver.cs` `UpdateAsync()` method, along with the other checks for valid Names etc.